src/precice/tests/WatchIntegralTest.cpp

```diff
@@ -334,7 +299,7 @@ BOOST_AUTO_TEST_CASE(VectorDataEdgeConnectivityNoScale)
   mesh->createEdge(v2, v3);
 
   PtrData doubleData = mesh->createData("DoubleData", 2, 0_dataID);
-  doubleData->setSampleAtTime(0, time::Sample(1, Eigen::VectorXd{{1., 2., 3., 4., 5., 6.}}));
+  doubleData->setSampleAtTime(0, time::Sample(2, Eigen::VectorXd{{1., 2., 3., 4., 5., 6.}}));
```

Comment:
Why not something similar to `mesh->data(1)->getDimensions()` like above?

Alternatively:

```
const int dataDim = 2
PtrData doubleData = mesh->createData("DoubleData", dataDim, 0_dataID);
  doubleData->setSampleAtTime(0, time::Sample(dataDim, Eigen::VectorXd{{1., 2., 3., 4., 5., 6.}}));
```

I think this makes the relationship much clearer and should also be safer for future changes.

Reply:
True. I'll have a look before merging